src/compute/src/sink/copy_to_s3_oneshot.rs

```diff
@@ -58,22 +57,29 @@ where
             }
         };
 
-        // Move all the data to a single worker and consolidate
-        // TODO(#7256): split the data among worker to some number of buckets (either static or configurable
-        // by the user) to split the load across the cluster.
-        let scope = sinked_collection.scope();
-        let active_worker = usize::cast_from(sink_id.hashed()) % scope.peers();
-        let exchange = Exchange::new(move |_| u64::cast_from(active_worker));
-        let error_exchange = Exchange::new(move |_| u64::cast_from(active_worker));
+        // Splitting the data across a known number of buckets to distribute load across the cluster.
+        // Each worker will be handling data belonging to 0 or more buckets. We are doing this so that
+        // we can write files to s3 deterministically across different replicas of different sizes
+        // using the bucket ID. Each worker will split a bucket's data into 1 or more
+        // files based on the user provided `MAX_FILE_SIZE`.
+        // TODO(#7256): make the number of buckets configurable.
+        let bucket_count = 20;
```

Comment:
I will try to get this in by tomorrow in this PR. But opening the PR now to get early feedback.

Reply:
done

Reply:
Will try to get the max number of workers across replicas instead
https://github.com/MaterializeInc/materialize/commit/4d0fd27e976ad851bf91c5a89bb7a245510eb55a#r139476753